### PR TITLE
change: remove ID of consumer

### LINKED ISSUE #1742: change: remove ID of consumer
# Feature request

## Please describe your feature

APISIX has removed the consumer id in version 2.5. 

Dashboard also needs to remove the consumer id before it can be used with APISIX 2.5 and above.

## Describe the solution you'd like

1. Sync schema from APISIX 2.5 version
2. Remove the id field from the consumer's struct
3. Modify related test cases

### LINKED ISSUE #1743: change: remove `listen.host` from api/conf/conf.yaml
# Feature request

## Please describe your feature

The configuration item `listen.host` in conf.yaml often confuses users, because when the configuration dashboard can be accessed non-locally, two places need to be modified (the other place is `allow_list`).
We can remove the `listen.host` configuration item from conf.yaml, but retain the feature, and users can add it when they need it.

## Describe the solution you'd like

1. remove `listen.host` from api/conf/conf.yaml
2. set `ServerHost` in [the code](https://github.com/apache/apisix-dashboard/blob/master/api/internal/conf/conf.go#L47
) to an empty string

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

APISIX has removed the consumer id in version 2.5, see: https://github.com/apache/apisix/pull/3868. 

Dashboard also needs to remove the consumer id before it can be used with APISIX 2.5 and above.

**What changes will this PR take into?**

1. Sync schema from APISIX 2.5 version
2. Remove the id field from the consumer's struct
3. Modify related test cases

**Related issues**

close #1742

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/test/e2e/route_with_plugin_jwt_test.go [hunk 1]
```diff
@@ -100,6 +101,12 @@ func TestRoute_With_Jwt_Plugin(t *testing.T) {
 	assert.Equal(t, http.StatusOK, status)
 	jwtToken := string(body)
 
+
+	// sleep for process log
+	time.Sleep(1500 * time.Millisecond)
+	logContent := ReadAPISIXErrorLog(t)
+	fmt.Println("logContent:", logContent)
```
Comment (starsz) [inline api/test/e2e/route_with_plugin_jwt_test.go]: This is debugging info?

File: api/test/docker/apisix_config.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ etcd:
     - "http://172.16.238.10:2379"
     - "http://172.16.238.11:2379"
     - "http://172.16.238.12:2379"
-  resync_delay: 0.1  # sync data from etcd quickly for e2e test
+  resync_delay: 1  # sync data from etcd quickly for e2e test
```
Comment (starsz) [inline api/test/docker/apisix_config.yaml:25]: Do we need to modify this?

File: api/internal/handler/consumer/consumer_test.go [hunk 3]
```diff
@@ -391,10 +369,12 @@ func TestHandler_Update(t *testing.T) {
 			mStore.On("Update", mock.Anything, mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
 				methodCalled = true
 				assert.Equal(t, tc.giveCtx, args.Get(0))
-				assert.Equal(t, tc.wantInput, args.Get(1))
 				assert.True(t, args.Bool(2))
 			}).Return(tc.giveRet, tc.giveErr)
 
+			mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+			}).Return(nil, nil)
```
Comment (starsz) [inline api/internal/handler/consumer/consumer_test.go:376]: We should test if the Get returns a consumer.

**Review by starsz — COMMENTED** [review]
It seems we don't modify E2E test.
Do we need?

**Comment by nic-chen** [comment]
@juzhiyuan @LiteSun  
FE CI failed, please help to check, thanks.

File: api/test/docker/apisix_config.yaml (hunk 2 above)
Comment (nic-chen) [reply api/test/docker/apisix_config.yaml]: sure. APISIX allow int only now.

File: api/test/e2e/route_with_plugin_jwt_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_jwt_test.go]: yes, removed.

**Comment by juzhiyuan** [comment]
@nic-chen Please take a look #1752, once it's merged, please resync your branch. Thanks!

**Comment by nic-chen** [comment]
> @nic-chen Please take a look #1752, once it's merged, please resync your branch. Thanks!

After synchronizing the code, it still failed. Does FE need to make responsive changes?

**Comment by juzhiyuan** [comment]
This test failed due to `raw data editor` 🤔 It's the first time that failed, if it still failed after rerun, then need @LiteSun's help.

**Comment by juzhiyuan** [comment]
https://user-images.githubusercontent.com/2106987/114357713-aaf42180-9ba4-11eb-9cbd-55a632395a47.mp4

@nic-chen I see what happened, the test failed due to Network error, does this PR changes some fields or?

**Comment by juzhiyuan** [comment]
see `00:34`

**Comment by nic-chen** [comment]
> does this PR changes some fields or?

remove ID of consumer. @juzhiyuan

**Review by tokers — APPROVED** [review]

**Comment by nic-chen** [comment]
FE CI still failed, please have a look when you have time. thanks. @LiteSun

**Comment by juzhiyuan** [comment]
https://github.com/apache/apisix-dashboard/pull/1736/commits/cecb6cff186cd9477fed19c554dfc4ffc7139d62

Wait for this PR https://github.com/apache/apisix-dashboard/pull/1736/commits/cecb6cff186cd9477fed19c554dfc4ffc7139d62 or take the above change :)

**Comment by juzhiyuan** [comment]
@LiteSun You may need to take some changes with FE & API.

File: api/test/docker/apisix_config2.yaml [hunk 4]
```diff
@@ -22,7 +22,7 @@ etcd:
     - "http://172.16.238.10:2379"
     - "http://172.16.238.11:2379"
     - "http://172.16.238.12:2379"
-  resync_delay: 0.1  # sync data from etcd quickly for e2e test
+  resync_delay: 0  # sync data from etcd quickly for e2e test
```
Comment (starsz) [inline api/test/docker/apisix_config2.yaml:25]: 0 is ok？

Comment (nic-chen) [reply api/test/docker/apisix_config2.yaml:25]: sure, you could see:
https://github.com/apache/apisix/blob/6a986f46490fa16166b83513ecde85555b6fe1b2/apisix/cli/ops.lua#L257
https://github.com/apache/apisix/blob/92b352862a7770daa0b1ac50893c4c68d84af877/apisix/core/config_etcd.lua#L593-L596
and
https://github.com/apache/apisix/blob/92b352862a7770daa0b1ac50893c4c68d84af877/apisix/core/config_etcd.lua#L563-L574

if we set 1 , will cause unstable.

**Review by gxthrj — APPROVED** [review]

File: api/test/docker/apisix_config2.yaml (hunk 4 above)
Comment (nic-chen) [reply api/test/docker/apisix_config2.yaml:25]: ping @starsz

Comment (starsz) [reply api/test/docker/apisix_config2.yaml:25]: Got it.Thank you for your explanation.

**Review by starsz — APPROVED** [review]

**Pull request merged**